presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -199,7 +199,23 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Set<Symbol>> context)
             PlanNode left = context.rewrite(node.getLeft(), leftInputs);
             PlanNode right = context.rewrite(node.getRight(), rightInputs);
 
-            return new JoinNode(node.getId(), node.getType(), left, right, node.getCriteria(), node.getFilter(), node.getLeftHashSymbol(), node.getRightHashSymbol());
+            List<Symbol> outputSymbols;
+            if (node.isNestedLoopJoin()) {
+                // do not prune nested joins output since it is not supported
+                outputSymbols = ImmutableList.<Symbol>builder()
```

Comment:
I haven't added output columns selection to nested loop join. I think nested loop join requires distributed mode in the first place.

I will add TODO to remove this IF when nested loop join supports output columns selection